plugins/hydra_optuna_sweeper/hydra_plugins/hydra_optuna_sweeper/config.py

```diff
@@ -0,0 +1,88 @@
+# Copyright (c) Facebook, Inc. and its affiliates. All Rights Reserved
+from dataclasses import dataclass, field
+from enum import Enum
+from typing import Any, Dict, List, Optional, Union
+
+from hydra.core.config_store import ConfigStore
+
+
+class DistributionType(Enum):
+    int = 1
+    float = 2
+    categorical = 3
+
+
+class Direction(Enum):
+    minimize = 1
+    maximize = 2
+
+
+@dataclass
+class DistributionConfig:
+
+    # Type of distribution. "int", "float" or "categorical"
+    type: DistributionType
+
+    # Choices of categorical distribution
+    choices: Optional[List[Union[str, int, float]]] = None
+
+    # Lower bound of int or float distribution
+    low: Optional[float] = None
+
+    # Upper bound of int or float distribution
+    high: Optional[float] = None
+
+    # If True, space is converted to the log domain
+    # Valid for int or float distribution
+    log: bool = False
+
+    # Discritization step
+    # Valid for int or float distribution
+    step: Optional[float] = None
+
+
+@dataclass
+class OptunaConfig:
+
+    # Direction of optimization
+    direction: Direction = Direction.minimize
+
+    # Storage URL to persist optimization results
+    # For example, you can use SQLite if you set 'sqlite:///example.db'
+    # Please refer to the reference for further details
+    # https://optuna.readthedocs.io/en/stable/reference/storages.html
+    storage: Optional[str] = None
+
+    # Name of study to persist optimization results
+    study_name: Optional[str] = None
+
+    # Total number of function evaluations
+    n_trials: int = 20
+
+    # Number of parallel workers
+    n_jobs: int = 2
+
+    # Sampling algorithm such as RandomSampler, TPESampler and CmaEsSampler
+    # Please refer to the reference for further details
+    # https://optuna.readthedocs.io/en/stable/reference/samplers.html
+    sampler: str = "TPESampler"
```

Comment:
I applied the Enum to the sampler type in https://github.com/facebookresearch/hydra/pull/1132/commits/3c712e48e67067fb735783f251f6eab206a50339.

I originally used `str` because we don't have to update the plugin code when new samplers are added to Optuna. But, the Enum approach is more helpful for users, and I applied your suggestion. Thank you for your suggestion.